packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -91,21 +92,38 @@ export type ISerializedStateManagerDocumentStorageService = Pick<
 
 export class SerializedStateManager {
 	private readonly processedOps: ISequencedDocumentMessage[] = [];
-	private snapshot: ISnapshotInfo | undefined;
 	private readonly mc: MonitoringContext;
+	private snapshot: ISnapshotInfo | undefined;
 	private latestSnapshot: ISnapshotInfo | undefined;
 	private refreshSnapshot: Promise<void> | undefined;
+	private snapshotFetchedTime: number | undefined;
+	private latestSnapshotFetchedTime: number | undefined;
+	private readonly lastSavedOpSequenceNumber: number | undefined;
+	private allSavedOpsProcessed: boolean;
 
 	constructor(
 		private readonly pendingLocalState: IPendingContainerState | undefined,
 		subLogger: ITelemetryLoggerExt,
 		private readonly storageAdapter: ISerializedStateManagerDocumentStorageService,
 		private readonly _offlineLoadEnabled: boolean,
+		containerEvent: EventEmitter,
```

Comment:
this need better typing. at a minimum it can reuse the typing IContainer uses to define its types